MoreLinq.Test/TestingSequence.cs

```diff
@@ -51,59 +63,262 @@ public enum Options
     /// </summary>
     sealed class TestingSequence<T> : IEnumerable<T>, IDisposable
     {
-        bool? _disposed;
-        IEnumerable<T>? _sequence;
+        private IEnumerable<T>? _sequence;
+        private readonly Options _options;
+        private readonly int _numEnumerations;
+
+        private bool _hasEnumerated;
+        private bool _currentlyEnumerating;
+        private int _disposedCount;
+        private int _enumerationCount;
 
-        internal TestingSequence(IEnumerable<T> sequence) =>
+        internal TestingSequence(IEnumerable<T> sequence, Options options, int numEnumerations)
+        {
             _sequence = sequence;
+            _numEnumerations = numEnumerations;
+            _options = options;
+        }
 
-        public bool IsDisposed => _disposed ?? false;
-        public bool IsReiterationAllowed { get; init; }
         public int MoveNextCallCount { get; private set; }
+        public bool IsDisposed => !_currentlyEnumerating;
```

Comment:
Tests were only written to validate the exception conditions. If there are other non-exception conditions that should be tested, then we can add tests for this and those conditions in a separate PR. However, my concern in writing the tests I did write was to confirm that we are validating the incorrect usages of enumerables.

Reply:
I understand but it helps to avoid regressions in refactoring the property implementation based on the new guts of the class. Since you went the distance to test the exception conditions, it would be a shame to leave this one out just.

Reply:
Ironic that you're usually the one asking for a _smaller_ PR. :-P

Addressed in 7a10be8.

Reply:
> Ironic that you're usually the one asking for a _smaller_ PR. :-P

Not asking for a smaller PR specifically but for changes in a PR to be as coherent as possible. This PR refactors the guts of `TestingSequence<T>` that's pretty fundamental to all tests. Adding tests was a very good call on your part because it not only gives us confidence that the refactored code works but also allows future contributors to refactor without fear. This property was touched by the refactoring and so it makes sense to cover it too.